src/main/scala/io/qbeast/core/model/QbeastSnapshot.scala

```diff
@@ -24,12 +26,26 @@ import org.apache.spark.sql.Dataset
  */
 trait QbeastSnapshot {
 
+  /**
+   * Returns the current snapshot object, representing the dataset state.
+   */
+  def snapshot: Any
+
   /**
    * The current state of the snapshot.
-   * @return
    */
   def isInitial: Boolean
 
+  /**
+   * Returns the total number of data files in the snapshot.
+   */
+  def allFilesCount: Long
+
+  /**
+   * Provides the schema of the dataset for this snapshot.
+   */
+  def schema: StructType
```

Comment:
It is good to return StructType instead of general Schema types, as we have parametrized in the in the QbeastCoreContext[DATA, DataSchema, QbeastOptions, FileDescriptor]. Maybe is best to change also the QbeastCoreContext and set Data as Dataframe and DataSchema to StructType, for coherence. ..
Or we do the opposite, and you make this class parametrized

Reply:
I will address this in an upcoming PR, as it will require more file changes, and I want to avoid this issue growing over time